src/libraries/System.Data.Common/src/System/Data/Common/ObjectStorage.cs

```diff
@@ -226,73 +226,89 @@ public override bool IsNull(int record)
         public override void Set(int recordNo, object value)
         {
             Debug.Assert(null != value, "null value");
+
             if (_nullValue == value)
             {
                 _values[recordNo] = null;
+                return;
```

Comment:
What is the aim behind replacing `else` blocks with many duplicated `return`s?

Reply:
To use a single throw at the end of the method whilst keeping the control flow simple. The JIT will create a single exit.

If this is less readable I can revert to `else` blocks.

Reply:
@MihaZupan What is your view on this?